senza/manaus/route53.py

```diff
@@ -35,12 +60,61 @@ def from_boto_dict(cls, hosted_zone_dict: Dict[str, Any]) -> 'Route53HostedZone'
         config = hosted_zone_dict['Config']
         resource_record_set_count = hosted_zone_dict['ResourceRecordSetCount']
 
-        def __repr__(self):
-            return '<Route53HostedZone: {name}>'.format_map(vars(self))
-
         return cls(id, name, caller_reference, config,
                    resource_record_set_count)
 
+    def delete(self,
+               changed_records: Iterable["Route53Record"],
+               comment: Optional[str] = None) -> Dict[str, Any]:
+        """
+        Submits changes to be deleted in Route53. Returns the dict sent to
+        AWS.
+
+        See:
+        http://boto3.readthedocs.io/en/latest/reference/services/route53.html#Route53.Client.change_resource_record_sets
+        """
+
+        client = boto3.client('route53')
+
+        change_batch = {'Changes': []}
+        if comment is not None:
+            change_batch['Comment'] = comment
+        for record in changed_records:
+            change = {'Action': 'DELETE',
+                      'ResourceRecordSet': record.boto_dict}
+            change_batch['Changes'].append(change)
+
+        client.change_resource_record_sets(HostedZoneId=self.id,
+                                           ChangeBatch=change_batch)
+
+        return change_batch
+
+    def upsert(self,
+               changed_records: Iterable["Route53Record"],
+               comment: Optional[str]=None) -> Dict[str, Any]:
+        """
+        Submits changes to be upsterted in Route53. Returns the dict sent to
+        AWS.
+
+        See:
+        http://boto3.readthedocs.io/en/latest/reference/services/route53.html#Route53.Client.change_resource_record_sets
+        """
+
+        client = boto3.client('route53')
+
+        change_batch = {'Changes': []}
+        if comment is not None:
+            change_batch['Comment'] = comment
+        for record in changed_records:
+            change = {'Action': 'UPSERT',
```

Comment:
Maybe we should extract this method behaviour? It is duplicated (like `delete`) and the only difference is the 'UPSERT' and 'DELETE' values in the "change" `dict`.